dio/lib/src/options.dart

```diff
@@ -336,52 +343,55 @@ class _RequestConfig {
     this.method,
     this.receiveTimeout,
     this.sendTimeout,
-    Map<String, dynamic> extra,
-    Map<String, dynamic> headers,
-    String contentType,
+    Map<String, dynamic>? extra,
+    Map<String, dynamic>? headers,
+    String? contentType,
     this.responseType,
     this.validateStatus,
-    this.receiveDataWhenStatusError = true,
-    this.followRedirects = true,
-    this.maxRedirects = 5,
+    bool? receiveDataWhenStatusError,
+    bool? followRedirects,
+    int? maxRedirects,
     this.requestEncoder,
     this.responseDecoder,
   }) {
     this.headers = headers ?? {};
     this.extra = extra ?? {};
```

Comment:
If you move `headers` and `extra` to the initializer list they don't have to be `late`.